### PR TITLE
feat: Add func metadata

### PR DESCRIPTION
**Pull request by dsimansk**
# Changes

- 🎁  feat: Add func metadata

/cc @lance @zroubalik

### CODE REVIEW

File: plugins/func/artifacthub-pkg.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Artifact Hub package metadata file
+# See https://github.com/artifacthub/hub/blob/master/docs/metadata/artifacthub-pkg.yml
+version: 1.9.0
+name: kn-plugin-func
```
Comment (dsimansk) [inline plugins/func/artifacthub-pkg.yml:18]: This name is just ID, but might be `func` only.

Comment (lance) [reply plugins/func/artifacthub-pkg.yml:18]: Evergreen question 🌲 
I prefer `func` but am OK either way

File: plugins/func/artifacthub-pkg.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+displayName: Knative Functions
```
[the remark this answers is not in the document]
Comment (lance) [reply plugins/func/artifacthub-pkg.yml:19]: I think I like having "Knative Functions" here to differentiate

**Review by lance — COMMENTED** [review]
/lgtm

**Pull request merged**